pkgs/top-level/all-packages.nix

```diff
@@ -1504,7 +1504,6 @@ with pkgs;
 
   collectd = callPackage ../tools/system/collectd {
     libmysql = mysql.lib;
-    libsigrok = libsigrok-0-3-0; # not compatible with >= 0.4.0 yet
```

Comment:
This is the preferred style: overriding versions in `all-packages` rather than specifying concrete versions in `default.nix`. Please keep it.

Reply:
I do not think ```collectd``` is such a case when it should be preferred.

  * For: customization. unlike ```libmysql``` which might be replaced with ```mariadb.client``` ```libsigrok``` is not a param to be customized here, ```collectd``` requires an obsolete version and it is immanent to ```collectd```, not to the configuration.

  * Against: ```collectd``` is often customized by reducing list of params and thus reduce the closure size (which is about 1Gb with all plugin and its deps).

It is common to have in ```/etc/nixos/configuration.nix``` something like:
 ```nix
callPackage ../tools/system/collectd {
  jdk = null;
  /* .... dozen of other params but ```libsigrok``` is forgotten */
}
``` 
Forgotten ```libsigrok``` would result in picking up ```pkgs.libsigrok``` which is of unsupported version 0.5

if you insist on that style, then maybe the param ```libsigrok``` should be renamed to something unique in order to prevent its unexpected defaulting to ```pkgs.libsigrok```

Reply:
> `callPackage ../tools/system/collectd`

I have not seen this form of overriding before, I doubt that it is common, and it certainly should not be used. Overriding is documented in Nixpkgs manual: https://nixos.org/nixpkgs/manual/#sec-overrides . The correct form is `pkgs.collectd.override { jdk = null; }`.

Please, keep this line as is.